src/main/java/com/openshift/jenkins/plugins/pipeline/NameValuePair.java

```diff
@@ -0,0 +1,34 @@
+package com.openshift.jenkins.plugins.pipeline;
+
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public class NameValuePair extends AbstractDescribableImpl<NameValuePair> {
+
+    protected final String name;
+    protected final String value;
+
+    @DataBoundConstructor
+    public NameValuePair(String name, String value) {
+        this.name = name;
+        this.value = value;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends Descriptor<NameValuePair> {
+        public String getDisplayName() {
+            return "Name/Value Pairs";
+        }
```

Comment:
Hey - I see where applyEnvVar trims and checks if the name is not empty before storing (don't remember if that was there before or not).  Was that your attempt to address this comment?

I was suggestion that you add validation methods to the DescriptorImpl for the name/value pair field (there are example in OpenShiftBuilder) and if you find issues, you can notify the user while he is constructing the build step.